src/Draco.Compiler/Internal/Symbols/Source/SourceModuleSymbol.cs

```diff
@@ -24,13 +26,19 @@ internal sealed class SourceModuleSymbol : ModuleSymbol, ISourceSymbol
     public override Symbol? ContainingSymbol { get; }
     public override string Name => this.declaration.Name;
 
-    public override SyntaxNode? DeclaringSyntax => null;
+    public override SymbolDocumentation Documentation => InterlockedUtils.InitializeNull(ref this.documentation, () => MarkdownDocumentationExtractor.Extract(this.RawDocumentation, this));
```

Comment:
Why take both the symbol and its documentation? If it's because it's private, make it `internal`, but no need to pass multiple things in from the same thing